### ISSUE TITLE
New icu_* crate releases break (unlocked) main build and latest crates release

### ISSUE THREAD

**Issue by mattsse** [boa-dev/boa#3299]
**Describe the bug**
0.17.0 on crates is currently broken due to new icu releases (icu_provider 0.13.0) which bumped zerovec through tinystr bump

**To Reproduce**
cargo new repro

add deps:

```toml
[dependencies]
boa_engine = "0.17"
```

`cargo c`

This is also an issue on main after removing lock file which then picks up new icu_* crates `rm Cargo.lock && cargo c`

**Comment by jedel1043** [comment]
Can you check if the build is now fixed? I think this broke because ICU4X published its crates by parts, instead of all at once.

**Comment by mattsse** [comment]
for new crates via `boa_engine = "0.17.0"` this is still the case because it pulls in all icu_ 1.3.0 crates

on main as well after rm Cargo.lock

**Comment by mattsse** [comment]
I guess https://github.com/boa-dev/boa/pull/3306 addresses this?

**Comment by jedel1043** [comment]
> I guess #3306 addresses this?

No, that's for the next version of Boa. We'll have to backport a fix and yank 0.17.

**Comment by jedel1043** [comment]
The problem is that we're pinning `zerovec` to "0.9.4" on `boa_icu_provider` without pinning the rest of the libraries to 1.2.0. I'll see if we can fix this on `boa_icu_provider` only. Otherwise, we can yank both `boa_engine@0.17` and `boa_icu_provider@0.17`

**Comment by Borber** [comment]
I still can't seem to find a solution. Is there any solution that can be solved now?

**Comment by jedel1043** [comment]
> I still can't seem to find a solution. Is there any solution that can be solved now?

We're preparing a new release that should publish at most tomorrow :)

**Comment by sffc** [comment]
tl;dr, icu 1.2 needs zerovec 0.9 and icu 1.3 needs zerovec 0.10, and mixing versions will cause errors.

**Comment by sffc** [comment]
Worth noting that icu 1.3 supports a range dependency such that it can work with zerovec 0.9.

**Comment by sffc** [comment]
> Can you check if the build is now fixed? I think this broke because ICU4X published its crates by parts, instead of all at once.

We hit a crates.io rate limit when publishing crates. We have asked for a rate limit increase so that next time we can shorten the amount of time it takes to ship a release on crates.io

**Comment by jedel1043** [comment]
> Worth noting that icu 1.3 supports a range dependency such that it can work with zerovec 0.9.

Like using a range version in Cargo.toml? Or what do you mean?

**Comment by jedel1043** [comment]
Ah, I see. ICU4X's repo pins zerovec to ">0.9.4, <0.11.0"
https://github.com/unicode-org/icu4x/blob/63a0b286b88d4e721530177316a5d8b518183c99/Cargo.toml#L217

**Comment by jedel1043** [comment]
Boa@0.17.1 is up! https://crates.io/crates/boa_engine/0.17.1
